### DISCUSSION TITLE
add a loop option for shell commands

### DISCUSSION THREAD

**Discussion by foadrezek** [zephyrproject-rtos/zephyr#78714] [Ideas]
I find myself running some shell commands multiple consecutive times, and thought of adding a loop mechanism, something like:
loop <iter> <cmd with args>

1- Is there a way to leverage the shell_execute_cmd shell API? As the documentation specificies, this cannot be done from a shell context (i.e in the loop handler) 
2- Is there a way to extract the handler of an shell command and call it directly ?
3- Use some shared memory/static variables for this purpose ? (i.e map the cmd name to the handler, later on the loop command can pick up this info and call the right handler)

**Comment by markushie** [comment]
Hi!
I think that's a great idea.

Is there a linux bash "watch" command equivalent in zephyr?
